[ROCm] Updating Stream Executor Blas interface (for ROCm) to call rocblas APIs…

deven-amd opened the pull request:
… that have been added in recent ROCm versions.

The ROCm implementation for the SE BLAS interface had a lot of the routines stubbed out because the underlying rocblas APIs were not yet implemented (at the time). Recent rocblas versions have been adding support for those BLAS APIs and this commit hooks them up.


------------------------------------------------


/cc @cheshire @chsigg @nvining-work @hawkinsp @inailuig

deven-amd committed:
Updating Stream Executor Blas interface (for ROCm) to call rocblas APIs that have been added in recent ROCm versions.

The ROCm implementation for the SE BLAS interface had a lot of the routines stubbed out because the underlying rocblas APIs were not yet implemented (at the time). Recent rocblas versions have been adding support for those BLAS APIs and this commit hooks them up.

hawkinsp commented on tensorflow/stream_executor/rocm/rocm_blas.cc:
```diff
@@ -460,17 +460,17 @@ bool ROCMBlas::DoBlasAsum(Stream *stream, uint64 elem_count,
-             << "for the \"complex<float>\" datatype";
-  return false;
+  return DoBlasInternal(wrap::rocblas_scasum, stream,
+                        false /* = pointer_mode_host */, elem_count,
```
As a general rule, TF style for named argument comments is:
`/*pointer_mode_host=*/false`

Same everywhere.

deven-amd replied:
pushed out a commit to address this

hawkinsp reviewed (changes requested):
One small style nit, but otherwise this seems like an uncontroversial change to me.

deven-amd committed:
cosmetic change to address code-review comments

changing all instance of the argument
from : `..., true /* = pointer_mode_host */, ...`
to   : `..., /* pointer_mode_host = */ true, ...`
